packages/cli/src/commands/verify.ts

```diff
@@ -0,0 +1,132 @@
+  rpcBatch: {
+    type: "boolean",
+    desc: "Enable batch processing of RPC requests in viem client (defaults to batch size of 100 and wait of 1s)",
+  },
+  srcDir: { type: "string", desc: "Source directory. Defaults to foundry src directory." },
+  verifier: { type: "string", desc: "The verifier to use" },
+  verifierUrl: {
+    type: "string",
+    desc: "The verification provider.",
+  },
+} as const;
+
+type Options = InferredOptionTypes<typeof verifyOptions>;
+
+const ERC1967_IMPLEMENTATION_SLOT = "0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc";
+
+const commandModule: CommandModule<Options, Options> = {
+  command: "verify",
+
+  describe: "Verify contracts",
+
+  builder(yargs) {
+    return yargs.options(verifyOptions);
+  },
+
+  async handler(opts) {
+    const profile = opts.profile ?? process.env.FOUNDRY_PROFILE;
+
+    const configV2 = (await loadConfig(opts.configPath)) as WorldConfig;
+    const config = worldToV1(configV2);
+
+    const srcDir = opts.srcDir ?? (await getSrcDirectory(profile));
+    const outDir = await getOutDirectory(profile);
+
+    const rpc = opts.rpc ?? (await getRpcUrl(profile));
+    console.log(
+      chalk.bgBlue(
+        chalk.whiteBright(`\n Verifying MUD contracts${profile ? " with profile " + profile : ""} to RPC ${rpc} \n`),
+      ),
+    );
+
+    const client = createWalletClient({
+      transport: http(rpc, {
+        batch: opts.rpcBatch
+          ? {
+              batchSize: 100,
+              wait: 1000,
+            }
+          : undefined,
+      }),
+    });
+
+    const publicClient = createPublicClient({
+      transport: http(rpc, {
+        batch: opts.rpcBatch
+          ? {
+              batchSize: 100,
+              wait: 1000,
+            }
+          : undefined,
+      }),
+    });
+
+    const contractNames = getExistingContracts(srcDir).map(({ basename }) => basename);
+    const resolvedWorldConfig = resolveWorldConfig(config, contractNames);
+
+    const systems = Object.keys(resolvedWorldConfig.systems).map((name) => {
+      const contractData = getContractData(`${name}.sol`, name, outDir);
+
+      return {
+        name,
+        bytecode: contractData.bytecode,
+      };
+    });
+
+    // Get modules
+    const modules = config.modules.map((mod) => {
+      const contractData =
+        defaultModuleContracts.find((defaultMod) => defaultMod.name === mod.name) ??
+        getContractData(`${mod.name}.sol`, mod.name, outDir);
+
+      return {
+        name: mod.name,
+        bytecode: contractData.bytecode,
+      };
+    });
+
+    // If the proxy implementation storage slot is set on the World, the World was deployed as a proxy.
+    const proxyImplementation = await publicClient.getStorageAt({
+      address: opts.worldAddress as Address,
+      slot: ERC1967_IMPLEMENTATION_SLOT,
+    });
+    const useProxy = proxyImplementation !== zeroHash;
```

Comment:
can we move this logic into `verify`? then we don't need the extra `verify` option